src/Tools/dotnet-format/MSBuild/MSBuildWorkspaceFinder.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+// LICENSING NOTE: The license for this file is from the originating 
+// source and not the general https://github.com/dotnet/roslyn license.
+// See https://github.com/aspnet/DotNetTools/blob/261b27b70027871143540af10a5cba57ce07ff97/src/dotnet-watch/Internal/MsBuildProjectFinder.cs
+
+using Microsoft.CodeAnalysis.Tools.CodeFormatter;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.Tools.MSBuild
+{
+    internal class MSBuildWorkspaceFinder
+    {
+        private static readonly string s_dnxProjectExtension = ".xproj";
```

Comment:
💭 Probably don't need to support .xproj

Reply:
We use the constant to exclude those as potential project files.